Sources/Misc/Lock.swift

```diff
@@ -13,19 +13,39 @@
 
 import Foundation
 
+/// A lock abstraction over an instance of `NSLocking`
 internal final class Lock {
 
-    private let recursiveLock: NSRecursiveLock = {
-        let lock = NSRecursiveLock()
+    #if swift(>=5.7)
+    private typealias UnderlyingType = NSLocking & Sendable
+    #else
+    // `NSRecursiveLock` and `NSLock` aren't `Sendable` until iOS 16.0 / Swift 5.7
+    private typealias UnderlyingType = NSLocking
+    #endif
+
+    private let lock: UnderlyingType
+    private init(_ lock: UnderlyingType) { self.lock = lock }
+
+    /// Initializes an instance backed by a non-recursive `NSLock`
+    convenience init() {
```

Comment:
kinda wonder if it now makes sense to have something like `createNonRecursive` so it's very clear when you're using them and you don't miss the fact that there's a choice. 
or even `lock = Lock(.recursive)` and `lock = Lock(.nonRecursive)`

Reply:
what do you think?

Reply:
Oh yeah totally 👍🏻 I'll change this.

Reply:
Done!